ops/internal/manage/depsets.go

```diff
@@ -0,0 +1,230 @@
+package manage
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/depset"
+	"github.com/ethereum-optimism/superchain-registry/ops/internal/config"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	errInvalidActivationTime = errors.New("activation_time is before interop_time")
+	errDepsetLengths         = errors.New("inconsistent depset lengths")
+	errInconsistentDepsets   = errors.New("inconsistent depset values")
+	errMissingAddress        = errors.New("missing address")
+)
+
+type DepsetChecker struct {
+	lgr             log.Logger
+	addrs           config.AddressesJSON
+	diskChainCfgs   map[uint64]DiskChainConfig
+	processedChains map[uint64]bool
+	chainsProcessed int
+}
+
+func NewDepsetChecker(logger log.Logger, cfgs []DiskChainConfig, addrs config.AddressesJSON) (*DepsetChecker, error) {
+	dc := &DepsetChecker{
+		lgr:             logger,
+		addrs:           addrs,
+		diskChainCfgs:   make(map[uint64]DiskChainConfig),
+		processedChains: make(map[uint64]bool),
+	}
+
+	for _, cfg := range cfgs {
+		dc.diskChainCfgs[cfg.Config.ChainID] = cfg
+	}
+
+	logger.Info("loaded depset checker instance with chain configs", "numDiskCfgs", len(cfgs))
+	return dc, nil
+}
+
+func (dc *DepsetChecker) Check() error {
+	for _, cfg := range dc.diskChainCfgs {
+		if dc.processedChains[cfg.Config.ChainID] {
+			// already processed this chain
+			continue
+		}
+		if cfg.Config.Interop == nil {
+			// nothing to check for this chain
+			dc.lgr.Info("skipping chain with no interop config", "chainID", cfg.Config.ChainID)
+			dc.processedChains[cfg.Config.ChainID] = true
+			dc.chainsProcessed++
+			continue
+		}
+
+		// collect all chains in the depset, then process them
+		depsetCfgs := []DiskChainConfig{}
+		depset := cfg.Config.Interop.Dependencies()
+
+		for chainId := range depset {
+			chainIdUint64, ok := chainId.Uint64()
+			if !ok {
+				return fmt.Errorf("chain ID is not a uint64: %v", chainId)
+			}
+			depsetCfgs = append(depsetCfgs, dc.diskChainCfgs[chainIdUint64])
```

Comment:
Should we check the config exists in this map?

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1013/commits/24f4a3f4afacca61103674b0ac63a0d253a6aa9e